### ISSUE TITLE
Weird Http/Keep-Alive monitoring stats

### ISSUE THREAD

**Issue by rds76** [payara/Payara#1141]
We have enabled keep-alive in http protocol, but there is 0 CountHit in stats. Or it looks like keep-alive doesnt work.

```
CountTimeouts   31539 count Oct 19, 2016 10:11:09 AM    Oct 19, 2016 4:25:32 PM --  Number of keep-alive connections that timed out
MaxRequests 250 count   Oct 19, 2016 10:11:09 AM    Oct 19, 2016 10:11:09 AM    --  Maximum number of requests allowed on a single keep-alive connection
CountHits   0 count Oct 19, 2016 10:11:09 AM    Oct 19, 2016 10:11:09 AM    --  Number of requests received by connections in keep-alive mode
SecondsTimeouts 30 seconds  Oct 19, 2016 10:11:09 AM    Oct 19, 2016 10:11:09 AM    --  Keep-alive timeout value in seconds
CountConnections    0 count Oct 19, 2016 10:11:09 AM    Oct 19, 2016 10:11:09 AM    --  Number of connections in keep-alive mode
CountFlushes    0 count Oct 19, 2016 4:25:34 PM Oct 19, 2016 4:25:34 PM --  Number of keep-alive connections that were closed
CountRefusals   0 count Oct 19, 2016 10:11:09 AM    Oct 19, 2016 10:11:09 AM    --  Number of keep-alive connections that were rejected
```

**Comment by mikecroft** [comment]
I think the likely explanation here is that the monitoring is not working as expected. I know @Pandrex247 has done quite a bit of work with various monitoring statistics which give incorrect/unexpected values. Sometimes it's due to an error, sometimes it's working as designed, but the attribute is very confusingly named. There is also the issue #1176 which is for a similar problem of odd values returned by MBeans.

I would point out, though, that the other values from the list there would suggest that keep-alive may not be configured properly.

Could you outline the steps you took so we can try to reproduce?

**Comment by rds76** [comment]
Below are my settings for http keep-alive. I could understand some unacurate monitoring values, but that looks like a) missconfigured/non-working keep-alive b) non-working monitoring for keep-alive
As you can see, there are no monitor values at all, so I cannot check it works.
```
Max Connections: 250
Maximum number of requests per connection in keep-alive mode.  
Timeout: 30 Seconds
Maximum time a connection can be deemed as idle and kept in the keep-alive state. A value of -1 will disable it. 
```

**Comment by fturizo** [comment]
Greetings @rds76, I reproduced this bug with the latest release of Payara. Even after reducing the timeout to 2 seconds from 30 seconds, the counters remained at 0. I've brought this issue to the developer team so they can check if these monitoring statistics are working correctly.  I'll update you back when I have information on the progress of this bug


The internal ID of this issue is PAYARA-1256

**Comment by jo-ka** [comment]
As an additional information: by using Apache HttpClient and its pooling connection manager that may reuse connections, I verified that connections _are_ kept open. By default for 30 seconds, and if I set another value explicitly through the timeout-seconds attribute, that is also respected. Only a value of 0 is not treated correctly according to [this](https://java.net/jira/browse/GLASSFISH-21009) GlassFish bug report, instead of keeping connections open indefinitely, Payara closes them instantly. So I'd say keep-alive is working in principle and it must be the monitoring that's broken.
